src/components/checkbox-group/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState, memo } from "react";
+
+import type { CheckboxGroupProps, CheckboxValue } from "./checkbox-group.types";
+
+const CheckboxGroup: React.FC<CheckboxGroupProps> = ({
+  groupTitle = "",
+  checkboxes,
```

Comment:
nit: A minor suggestion I would make is to consider initializing the `checkboxes` prop with a default value (empty array), like so:

```jsx
checkboxes = [],
```

Reply:
Thank you for your input. I have initialized it now.